src/automations/maps/workQueueStatusTrigger.ts

```diff
@@ -0,0 +1,106 @@
+      ...toMatchRelatedId('work-pool', source.workPools),
+    },
+    forEach: ['prefect.resource.id'],
+    expect: [mapWorkQueueStatusToEvent(source.status)],
+  })
+}
+
+function mapProactiveWorkQueueStatusTriggerToAutomationTrigger(source: WorkQueueStatusTrigger): AutomationTriggerEvent {
+  return new AutomationTriggerEvent({
+    posture: 'Proactive',
+    match: {
+      'prefect.resource.id': toResourceId('prefect.work-queue', source.workQueues),
+    },
+    matchRelated: {
+      ...toMatchRelatedId('work-pool', source.workPools),
+    },
+    forEach: ['prefect.resource.id'],
+    expect: anyStatusExcept(source.status).map(mapWorkQueueStatusToEvent),
+    after: [mapWorkQueueStatusToEvent(source.status)],
+    within: source.time,
+  })
+}
+
+function anyStatusExcept(status: WorkPoolQueueStatus): WorkPoolQueueStatus[] {
+  return (['ready', 'not_ready', 'paused'] as const).filter(_status => _status !== status)
+}
+
+export const mapAutomationTriggerToWorkQueueStatusTrigger: MapFunction<AutomationTriggerEvent, WorkQueueStatusTrigger> = function(source) {
+  if (source.posture === 'Reactive') {
+    return mapReactiveAutomationTriggerToWorkQueueStatusTrigger(source)
+  }
+
+  return mapProactiveAutomationTriggerToWorkQueueStatusTrigger(source)
+}
+
+function mapReactiveAutomationTriggerToWorkQueueStatusTrigger(trigger: AutomationTriggerEvent): WorkQueueStatusTrigger {
+  return {
+    workPools: fromResourceId('prefect.work-pool', trigger.matchRelated['prefect.resource.id']),
+    workQueues: fromResourceId('prefect.work-queue', trigger.match['prefect.resource.id']),
+    status: statusFromAutomationTriggerEvent(trigger),
+    posture: 'Reactive',
+    time: trigger.within,
+  }
+}
+
+function mapProactiveAutomationTriggerToWorkQueueStatusTrigger(trigger: AutomationTriggerEvent): WorkQueueStatusTrigger {
+  return {
+    workPools: fromResourceId('prefect.work-pool', trigger.matchRelated['prefect.resource.id']),
+    workQueues: fromResourceId('prefect.work-queue', trigger.match['prefect.resource.id']),
+    status: statusFromAutomationTriggerEvent(trigger),
+    posture: 'Proactive',
+    time: trigger.within,
+  }
+}
+
+const statusEventToStatus: Record<WorkQueueStatusEvent, WorkPoolQueueStatus> = {
+  'prefect.work-queue.ready': 'ready',
+  'prefect.work-queue.not-ready': 'not_ready',
+  'prefect.work-queue.paused': 'paused',
+}
+
+function statusFromAutomationTriggerEvent(trigger: AutomationTriggerEvent): WorkPoolQueueStatus {
+  const statusEvents = trigger.posture === 'Reactive' ? trigger.expect : trigger.after
+
+  for (const statusEvent of statusEvents) {
+    if (isWorkQueueStatusEvent(statusEvent)) {
+      return statusEventToStatus[statusEvent]
+    }
+  }
+
+  throw new Error(`Unknown work queue status events: ${statusEvents}`)
+}
+
+function mapWorkQueueStatusToEvent(status: WorkPoolQueueStatus): WorkQueueStatusEvent {
+  switch (status) {
+    case 'not_ready':
+      return 'prefect.work-queue.not-ready'
```

Comment:
I forget, does this need a `not-ready` case as well?

Reply:
I believe this is correct but let me verify

Reply:
Verified with @collincchoy this is correct 👍

Reply:
Great, thank you 🙏🏻